src/api.rs

```diff
@@ -111,11 +124,26 @@ pub struct PlayersResponse {
 /// This is used by new host connections to update thier display to match the current state of the
 /// game.
 #[get("/players")]
-pub fn get_players(scoreboard: State<Mutex<Scoreboard>>) -> JSON<PlayersResponse> {
+pub fn get_players(
+    scoreboard: State<Mutex<Scoreboard>>,
+    usernames: State<Mutex<Usernames>>,
+) -> JSON<PlayersResponse> {
     // Clone the scoreboard so we can release the lock on it quickly.
     let scoreboard = scoreboard.lock().expect("Scoreboard mutex was poisoned").clone();
 
-    let players = scoreboard.iter().map(|(&id, &score)| PlayerData { id, score }).collect();
+    let usernames = usernames.lock().expect("Usernames mutex was poisoned").clone();
+
+    let players = scoreboard.iter()
+        .map(
+            |(&id, &score)|
+            {
+                let username = match usernames.get(&id) {
+                    None => "".to_string(),
+                    Some(i) => i.to_string(),
+                };
```

Comment:
What happens when the app `panic!`s? Execution ends with sending the message in `expect()`?

Side note: I dig the match `Option<T>` concept. Took a minute to get my head around it.